source/simulator/assembly.cc

```diff
@@ -779,14 +781,29 @@ namespace aspect
                                               (parameters.stokes_velocity_degree+1)/2),
                                   parameters.n_compositional_fields);
 
+    int count_smoothing = 0;
     typename DoFHandler<dim>::active_cell_iterator cell = dof_handler.begin_active();
     for (unsigned int cellidx=0; cellidx<triangulation.n_active_cells(); ++cellidx, ++cell)
       {
-        if (!cell->is_locally_owned())
-          {
-            viscosity_per_cell[cellidx]=-1;
-            continue;
-          }
+    	cell->set_user_index(count_smoothing);
```

Comment:
you need to skip artificial cells. Did you run your code in parallel to see if it works?